--- a/src/VBox/Runtime/generic/http-curl.cpp	2026-02-13 19:26:39.506334407 +0300
+++ b/src/VBox/Runtime/generic/http-curl.cpp	2026-02-13 19:27:52.350142975 +0300
@@ -696,7 +696,7 @@
     Log(("rtHttpUpdateAutomaticProxyDisable: pThis=%p\n", pThis));
 
     AssertReturn(curl_easy_setopt(pThis->pCurl, CURLOPT_PROXYTYPE,   (long)CURLPROXY_HTTP) == CURLE_OK, VERR_INTERNAL_ERROR_2);
-    pThis->enmProxyType = CURLPROXY_HTTP;
+    pThis->enmProxyType = (curl_proxytype)CURLPROXY_HTTP;
 
     AssertReturn(curl_easy_setopt(pThis->pCurl, CURLOPT_PROXYPORT,             (long)1080) == CURLE_OK, VERR_INTERNAL_ERROR_2);
     pThis->uProxyPort = 1080;
@@ -871,7 +871,7 @@
             curl_proxytype enmProxyType;
             if (RTUriIsSchemeMatch(pszProxyUrl, "http"))
             {
-                enmProxyType  = CURLPROXY_HTTP;
+                enmProxyType  = (curl_proxytype)CURLPROXY_HTTP;
                 if (uProxyPort == UINT32_MAX)
                     uProxyPort = 80;
             }
@@ -879,7 +879,7 @@
 # if CURL_AT_LEAST_VERSION(7,52,0)
             else if (RTUriIsSchemeMatch(pszProxyUrl, "https"))
             {
-                enmProxyType  = CURLPROXY_HTTPS;
+                enmProxyType  = (curl_proxytype)CURLPROXY_HTTPS;
                 if (uProxyPort == UINT32_MAX)
                     uProxyPort = 443;
             }
@@ -887,17 +887,17 @@
 #endif
             else if (   RTUriIsSchemeMatch(pszProxyUrl, "socks4")
                      || RTUriIsSchemeMatch(pszProxyUrl, "socks"))
-                enmProxyType = CURLPROXY_SOCKS4;
+                enmProxyType = (curl_proxytype)CURLPROXY_SOCKS4;
             else if (RTUriIsSchemeMatch(pszProxyUrl, "socks4a"))
-                enmProxyType = CURLPROXY_SOCKS4A;
+                enmProxyType = (curl_proxytype)CURLPROXY_SOCKS4A;
             else if (RTUriIsSchemeMatch(pszProxyUrl, "socks5"))
-                enmProxyType = CURLPROXY_SOCKS5;
+                enmProxyType = (curl_proxytype)CURLPROXY_SOCKS5;
             else if (RTUriIsSchemeMatch(pszProxyUrl, "socks5h"))
-                enmProxyType = CURLPROXY_SOCKS5_HOSTNAME;
+                enmProxyType = (curl_proxytype)CURLPROXY_SOCKS5_HOSTNAME;
             else
             {
                 fUnknownProxyType = true;
-                enmProxyType = CURLPROXY_HTTP;
+                enmProxyType = (curl_proxytype)CURLPROXY_HTTP;
                 if (uProxyPort == UINT32_MAX)
                     uProxyPort = 8080;
             }
@@ -1337,7 +1337,7 @@
      * Determine the proxy type (not entirely sure about type == proxy type and
      * not scheme/protocol)...
      */
-    curl_proxytype  enmProxyType      = CURLPROXY_HTTP;
+    curl_proxytype  enmProxyType      = (curl_proxytype)CURLPROXY_HTTP;
     uint32_t        uDefaultProxyPort = 8080;
     if (   CFEqual(hStrProxyType, kCFProxyTypeHTTP)
         || CFEqual(hStrProxyType, kCFProxyTypeHTTPS))
@@ -1346,7 +1346,7 @@
     {
         /** @todo All we get from darwin is 'SOCKS', no idea whether it's SOCK4 or
          *        SOCK5 on the other side... Selecting SOCKS5 for now. */
-        enmProxyType = CURLPROXY_SOCKS5;
+        enmProxyType = (curl_proxytype)CURLPROXY_SOCKS5;
         uDefaultProxyPort = 1080;
     }
     /* Unknown proxy type. */
@@ -2077,7 +2077,7 @@
      * leave that to cURL.  (A bit afraid of breaking user settings.)
      */
     pThis->fUseSystemProxySettings = false;
-    return rtHttpUpdateProxyConfig(pThis, CURLPROXY_HTTP, pcszProxy, uPort ? uPort : 1080, pcszProxyUser, pcszProxyPwd);
+    return rtHttpUpdateProxyConfig(pThis, (curl_proxytype)CURLPROXY_HTTP, pcszProxy, uPort ? uPort : 1080, pcszProxyUser, pcszProxyPwd);
 }
 
 
@@ -3594,7 +3594,7 @@
             pProxy->enmProxyType = RTHTTPPROXYTYPE_NOPROXY;
         else
         {
-            switch (pThis->enmProxyType)
+            switch ((long long)pThis->enmProxyType)
             {
                 case CURLPROXY_HTTP:
 #ifdef CURL_AT_LEAST_VERSION
