diff --git a/cowsay/cowsay b/cowsay/cowsay
index ce3bbd1..781d344 100755
--- a/cowsay/cowsay
+++ b/cowsay/cowsay
@@ -19,11 +19,13 @@ if (${^UTF8LOCALE}) {
     eval { $_ = Encode::decode_utf8($_,1) } for @ARGV;
 }
 
+$progpath = Cwd::realpath(__FILE__);
 $version = "3.03";
-$progname = basename($0);
+$progname = basename($progpath);
 $eyes = "oo";
 $tongue = "  ";
-$cowpath = $ENV{'COWPATH'} || '%PREFIX%/share/cows';
+$prefix = dirname($progpath);
+$cowpath = $ENV{'COWPATH'} || "$prefix/cows";
 @message = ();
 $thoughts = "";
 
diff --git a/cowsay/install.sh b/cowsay/install.sh
index 814331b..e784fa6 100755
--- a/cowsay/install.sh
+++ b/cowsay/install.sh
@@ -69,23 +69,17 @@ PREFIX=${prefix:-/usr/local}
 echo Okay, time to install this puppy.
 
 echo s,%BANGPERL%,!$usethisperl,\; > $cwd/install.pl
-echo s,%PREFIX%,$PREFIX,\; >> $cwd/install.pl
 set -x
-mkdir -p $PREFIX/bin || (mkdir $PREFIX; mkdir $PREFIX/bin)
-$usethisperl -p $cwd/install.pl $cwd/cowsay > $PREFIX/bin/cowsay
-chmod a+x $PREFIX/bin/cowsay
-ln -s cowsay $PREFIX/bin/cowthink
-mkdir -p $PREFIX/man/man1 || ($mkdir $PREFIX; mkdir $PREFIX/man; mkdir $PREFIX/man/man1)
-$usethisperl -p $cwd/install.pl $cwd/cowsay.1 > $PREFIX/man/man1/cowsay.1
-chmod a+r $PREFIX/man/man1/cowsay.1
-ln -s cowsay.1 $PREFIX/man/man1/cowthink.1
-mkdir -p $PREFIX/share/cows || (mkdir $PREFIX; mkdir $PREFIX/share; mkdir $PREFIX/share/cows)
-tar -C $cwd -cf - $filelist | (cd $PREFIX/share && tar -xvf -)
+mkdir -p $PREFIX
+$usethisperl -p $cwd/install.pl $cwd/cowsay > $PREFIX/cowsay
+chmod a+x $PREFIX/cowsay
+mkdir -p $PREFIX/cows
+tar -C $cwd -cf - $filelist | (cd $PREFIX && tar -xvf -)
 set +x
 
 echo Okay, let us see if the install actually worked.
 
-if [ ! -f $PREFIX/share/cows/default.cow ]; then
+if [ ! -f $PREFIX/cows/default.cow ]; then
     echo The default cow file did not make it across!
     echo Ooops, it failed...sorry!
     exit 1
diff --git a/jumoreski.sh b/jumoreski.sh
index 7eda414..aa18b8d 100644
--- a/jumoreski.sh
+++ b/jumoreski.sh
@@ -3,13 +3,13 @@ function print_motd() {
     # Run only in an interactive shell
     if [ ! -z "$PS1" ]; then
         fortune $datafile | if [ "`date +"%w"`" = "3" ]; then
-            cowsay -f small-frogs-wednesday -W 80
+            /usr/share/jumoreski/cowsay/cowsay -f small-frogs-wednesday -W 80
         else
-            cowsay -W 80
+            /usr/share/jumoreski/cowsay/cowsay -W 80
         fi | lolcat
     fi
 }
 
 function print_jumoreski() {
-    print_motd /usr/share/games/jumoreski
+    print_motd /usr/share/jumoreski/jumoreski
 }
